### PR TITLE
No more static IPs for the CyHy VPC

### PR DESCRIPTION
**Pull request by jsf9k**
I deployed this to the `jsf9k` workspace and it appears to be working fine.  I particularly want @felddy to review this pull request in case there is a good reason to stick with static IPs that I'm missing.

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
This all makes sense to me, but agree that @felddy should weigh in with his DNS expertise in case we are forgetting about something.

File: terraform/cyhy_private_dns.tf [hunk 1]
```diff
@@ -82,9 +82,9 @@ resource "aws_route53_record" "cyhy_vulnscan_A" {
 resource "aws_route53_zone" "cyhy_scanner_zone_reverse" {
   # NOTE:  This assumes that we are using /24 blocks
   name = "${format("%s.%s.%s.in-addr.arpa.",
-    element( split(".", aws_subnet.cyhy_scanner_subnet.cidr_block) ,2),
```
Comment (felddy) [inline terraform/cyhy_private_dns.tf:85]: Good catch... ?
How about that ") ,2)"

File: terraform/cyhy_private_dns.tf [hunk 2]
```diff
@@ -118,7 +118,12 @@ resource "aws_route53_record" "cyhy_rev_3_PTR" {
 
 resource "aws_route53_record" "cyhy_rev_bastion_PTR" {
   zone_id = "${aws_route53_zone.cyhy_scanner_zone_reverse.zone_id}"
-  name    = "${local.the_bastion}.${aws_route53_zone.cyhy_scanner_zone_reverse.name}"
+  name    = "${format("%s.%s.%s.%s.in-addr.arpa.",
```
Comment (felddy) [inline terraform/cyhy_private_dns.tf:121]: We'll have to revisit this for IPv6.

File: terraform/cyhy_private_dns.tf [hunk 3]
```diff
@@ -161,17 +176,26 @@ resource "aws_route53_zone" "cyhy_private_zone_reverse" {
...
+    element(split(".", aws_instance.cyhy_mongo.private_ip) ,3),
+    element(split(".", aws_instance.cyhy_mongo.private_ip) ,2),
+    element(split(".", aws_instance.cyhy_mongo.private_ip) ,1),
+    element(split(".", aws_instance.cyhy_mongo.private_ip) ,0),
+  )}"
   type    = "PTR"
   ttl     = 300
-  records = [ "database${count.index + 1}.${local.cyhy_private_domain}." ]
+  records = [ "database1.${local.cyhy_private_domain}." ]
```
Comment (felddy) [inline terraform/cyhy_private_dns.tf]: What happened to our multiple database support?

**Review by felddy — CHANGES_REQUESTED** [review]
I think there might be an issue with hard coding in `database1` in `cyhy_rev_database_PTR`.   We still have `count_database` defined in `locals.tf`

File: terraform/cyhy_private_dns.tf (hunk 1 above)
Comment (jsf9k) [reply terraform/cyhy_private_dns.tf:85]: Now fixed.

File: terraform/cyhy_private_dns.tf (hunk 2 above)
Comment (jsf9k) [reply terraform/cyhy_private_dns.tf:121]: Agreed, but that was true before too.

File: terraform/cyhy_private_dns.tf (hunk 3 above)
Comment (jsf9k) [reply terraform/cyhy_private_dns.tf]: It's back.

**Review by felddy — APPROVED** [review]
Is nice.

**Pull request merged**